code/modules/mob/living/silicon/ai/ai_defense.dm

```diff
@@ -61,14 +61,18 @@
 /mob/living/silicon/ai/flash_act(intensity = 1, override_blindness_check = 0, affect_silicon = 0, visual = 0, type = /atom/movable/screen/fullscreen/flash, length = 25)
 	return // no eyes, no flashing
 
-/mob/living/silicon/ai/emag_act(mob/user, obj/item/card/emag/emag_card)///emags access panel lock, so you can crowbar it without robotics access or consent
+/mob/living/silicon/ai/emag_act(mob/user, obj/item/card/emag/emag_card) ///emags access panel lock, so you can crowbar it without robotics access or consent
 	. = ..()
 	if(emagged)
-		balloon_alert(user, "access panel lock already shorted!")
+		if (user)
+			balloon_alert(user, "access panel lock already shorted!")
 		return
```

Comment:
```suggestion
		return FALSE
```
Should this return false here if we didn't do anything?

Reply:
its totally fine to return ., since the default return of atom/emag_act() is null, and if it is true for whatever reason, we still want to tell the caller "heyo the emag succeeded in some way"

ill eventually return to this proc and do a SHOULD_CALL_PARENT and ill probably change most return FALSE to return . for this very reaosn